src/ldc/eh/win32.d

```diff
@@ -233,22 +286,41 @@ nothrow:
         return _p[--_length];
     }
 
+    void shrink(size_t sz)
+    {
+        while (_length > sz)
+            _p[--_length] = null;
+    }
+
     ref inout(Throwable) opIndex(size_t idx) inout
     {
         return _p[idx];
     }
 
+    size_t find(Throwable e)
+    {
+        for (size_t i = _length; i > 0; )
+            if (exceptionStack[--i] is e)
+                return i;
+        return ~0;
+    }
+
     @property size_t length() const { return _length; }
     @property bool empty() const { return !length; }
 
+    void swap(ref ExceptionStack other)
+    {
+        auto olength = other._length; other._length = _length; _length = olength;
+        auto op      = other._p;      other._p      = _p;      _p = op;
+        auto ocap    = other._cap;    other._cap    = _cap;    _cap = ocap;

```

Comment:
Done. I think there should be a swap function somewhere in druntime, it is implemented again and again...